inc/class-rewrite.php

```diff
@@ -131,24 +131,36 @@ public function category_rewrite_rules() {
 		$categories = get_categories( array( 'hide_empty' => false ) );
 		if ( is_array( $categories ) && $categories !== array() ) {
 			foreach ( $categories as $category ) {
-				$category_nicename = $category->slug;
+				$category_nicename          = $category->slug;
+				$category_nicename_filtered = $this->filteredNicename( $category_nicename );
 				if ( $category->parent == $category->cat_ID ) {
 					// Recursive recursion.
 					$category->parent = 0;
 				}
 				elseif ( $taxonomy->rewrite['hierarchical'] != 0 && $category->parent !== 0 ) {
 					$parents = get_category_parents( $category->parent, false, '/', true );
 					if ( ! is_wp_error( $parents ) ) {
-						$category_nicename = $parents . $category_nicename;
+						$category_nicename          = $parents . $category_nicename;
+						//Convert encoded parts of the URI to Uppercase.
+						$category_nicename_filtered = $this->filteredNicename( $category_nicename );
 					}
 					unset( $parents );
 				}
 
 				$category_rewrite[ $blog_prefix . '(' . $category_nicename . ')/(?:feed/)?(feed|rdf|rss|rss2|atom)/?$' ]                = 'index.php?category_name=$matches[1]&feed=$matches[2]';
 				$category_rewrite[ $blog_prefix . '(' . $category_nicename . ')/' . $wp_rewrite->pagination_base . '/?([0-9]{1,})/?$' ] = 'index.php?category_name=$matches[1]&paged=$matches[2]';
 				$category_rewrite[ $blog_prefix . '(' . $category_nicename . ')/?$' ] = 'index.php?category_name=$matches[1]';
+				
+				//Add rules for the uppercase encoded URIs.
+				if ( $category_nicename_filtered !== $category_nicename ) {
+
+					$category_rewrite[ $blog_prefix . '(' . $category_nicename_filtered . ')/(?:feed/)?(feed|rdf|rss|rss2|atom)/?$' ]                = 'index.php?category_name=$matches[1]&feed=$matches[2]';
```

Comment:
You are repeating the code of 150-152. It would be nice to have a function for it, just to prevent 'duplicate' code:
```php
protected function add_category_rewrites( $rewrites, $category_name, $blog_prefix, $pagination_base ) {
    $rewrite_name = $blog_prefix . '(' . $category_name . ')';
				
	$rewrites[ $rewrite_name . '/(?:feed/)?(feed|rdf|rss|rss2|atom)/?$' ] = 'index.php?category_name=$matches[1]&feed=$matches[2]';
	$rewrites[ $rewrite_name . '/' . $pagination_base . '/?([0-9]{1,})/?$' ] = 'index.php?category_name=$matches[1]&paged=$matches[2]';
	$rewrites[ $rewrite_name . '/?$' ] = 'index.php?category_name=$matches[1]';

    return $rewrites;
}
```